torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -607,7 +829,50 @@ def generate_detector_report(self, model: GraphModule) -> Tuple[str, Dict[str, A
                 their s_c metric compared to the threshold
                 the threshold used to make the recommendation
                 the channel used for recording data
-                the input channel range ratio
-                the weight channel range ratio
+                the input channel range info
+                the weight channel range info
         """
-        pass
+
+        # find the range of inputs
+        input_values: Dict[str, Dict] = self._extract_input_info(model)
+
+        # find the range of weights
+        weight_values: Dict[str, Dict] = self._extract_weight_info(model)
+
+        # calculate per_channel comparision statistic s_c
+        comp_stats: Dict[str, torch.Tensor] = self._generate_comparision_values(input_values, weight_values)
+
+        # generate the return dictionary
+        input_weight_equalization_info: Dict[str, Dict] = self._generate_dict_info(input_values, weight_values, comp_stats)
+
+        # now we can generate report based on this information
+        input_weight_string = "Input-Weight Equalization suggestions: \n"
+
+        # some strings to be formatted depending on module we are adding
+        module_suggestion_str = "For Module {} looked at with axis {} we suggest: \n"
+        channel_suggestion_str = "\tFor channel {}, we suggest {} input weight equalization because {}\n"
+        use_str = "to use"
+        no_use_str = "to not use"
+        input_weight_benefit_str = "we expect significant reduction in quantization error."
+        input_weight_non_benefit_reasoning = "the scales of the input vs. weight with regards to their ranges."
+        input_weight_non_benefit_str = "we don't expect much improvement from input-weight equalization based on {}"
```

Comment:
Do we have any tests for these strings?

Reply:
There isn't a direct test for the strings themselves. However, all of them are formatted using values from the input_weight_equalization_info which is tested. The strings themselves are essentially just a boilerplate template that the dict populates, so the dict is what is primarily tested.